### ISSUE TITLE
emitter crashes with computed object literal names

### ISSUE THREAD

**Issue by jrieken** [Microsoft/TypeScript#8532]
**TypeScript Version:** 

1.8.x

**Code**

```ts
// A self-contained demonstration of the problem follows...
function createDefaultProperties(): { [name: string]: any } {

	const startTime = Date.now();
	let eventCounter = 0;

	return {
		get ['common.sequenceId']() { return eventCounter++; },
		get timestamp() { return new Date(); },
		get ['common.timesincesessionstart']() { return Date.now() - startTime; }
	};
}
```

* open playground website
* paste code above
* it crashes like so:

```
Object {stack: "TypeError: Cannot read property 'declarations' of …script/common/lib/typescriptServices.js:19:13310)", message: "TypeError: Cannot read property 'declarations' of undefined"}
message
:
"TypeError: Cannot read property 'declarations' of undefined"
stack
:
"TypeError: Cannot read property 'declarations' of undefined↵    at e.checkNode (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:1:30909)↵    at t.e._checkNodeOrToken (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:2:2107)↵    at t.e._visit (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:2:1801)↵    at https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:2:1853↵    at r (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/lib/typescriptServices.js:19:10966)↵    at Object.i [as forEachChild] (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/lib/typescriptServices.js:19:14002)↵    at t.e._visit (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:2:1824)↵    at https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/typescriptWorker2.js:2:1853↵    at a (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/lib/typescriptServices.js:19:11081)↵    at Object.i [as forEachChild] (https://www.typescriptlang.org/play/Script/vs/languages/typescript/common/lib/typescriptServices.js:19:13310)"
__proto__
:
Object
```

**Comment by RyanCavanaugh** [comment]
Fixed in master branch

**Comment by RyanCavanaugh** [comment]
Actually, this doesn't repro in 1.8 either. Do you have a command-line repro?

**Comment by RyanCavanaugh** [comment]
This is a Playground bug. It's crashing inside some code that checks for unused variables, but I can't find the source for this in either our repo or vscode's. Any idea where it comes from?

**Comment by jrieken** [comment]
I see the same with gulp-tsb and TypeScript 1.8.9

**Comment by jrieken** [comment]
the playground hint might be wrong... I added that for ease, but when using `1.8.10` it crashes like so:

```
TypeError: Cannot read property 'decorators' of undefined
    at getStartPos (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:30547:45)
    at emitStart (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:30551:21)
    at emitDownlevelObjectLiteralWithComputedProperties (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:33774:25)
    at emitObjectLiteral (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:33864:25)
    at emitJavaScriptWorker (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:39065:32)
    at emitNodeWithoutSourceMap (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:38953:21)
    at emitNodeWithSourceMap (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:38947:21)
    at emitNodeConsideringCommentsOption (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:38938:21)
    at emit (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:38920:17)
    at emitOptional (/Users/jrieken/Code/vscode/node_modules/typescript/lib/typescript.js:32800:21)
```

**Comment by RyanCavanaugh** [comment]
That's a completely different callstack from the one in the Playground.

It seems the latter crash requires the `--sourcemap` flag to be provided in order to repro, and does currently repro in master.

**Comment by andy-ms** [comment]
The first problem is due to some code in the playground (taken from vscode) which uses outdated const enums and should be fixed when we next update the playground. The second problem is fixed by #8577.
